lib/Utility/XMLResponse.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @copyright Copyright (c) 2016, ownCloud, Inc.
+ *
+ * @author Bernhard Posselt <dev@bernhard-posselt.com>
+ * @author Christoph Wurst <christoph@winzerhof-wurst.at>
+ * @author Lukas Reschke <lukas@statuscode.ch>
+ * @author Morris Jobke <hey@morrisjobke.de>
+ * @author Roeland Jago Douma <roeland@famdouma.nl>
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ * @author Thomas Tanghus <thomas@tanghus.net>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+namespace OCA\News\Utility;
+
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Response;
+
+/**
+ * A renderer for JSON calls
+ * @since 6.0.0
+ */
+class XMLResponse extends Response {
```

Comment:
Sadly I could not find one, only a JsonResponse. But I'm new to the codebase so I might have missed it.

Reply:
There is none. However, your implementation isn't really an XML response because the XML is already built outside ot the Response class.
https://docs.nextcloud.com/server/stable/developer_manual/basics/controllers.html#creating-custom-responses

How about creating a dedicated AtomResponse and move the FeedIo call into the render function?
```php
// response
class AtomResponse extends Response {
    public function __construct($feed) {
        $this->addHeader('Content-Type', 'application/atom+xml');
        $this->feed = $feed;
    }

    public function render() : string {
        return $this->feedIo->toAtom($this->feed);
    }
}

// controller
        $response = new AtomResponse($feed);
```

Alternatively you could try the existing `DataDisplayResponse`, but I'm not sure if the `Content-Disposition` header in there would interfere.